src/core/core.scale.js

```diff
@@ -1017,18 +1017,15 @@ var Scale = Element.extend({
 				continue;
 			}
 
-			if (i === me.zeroLineIndex && options.offset === offsetGridLines) {
-				// Draw the first index specially
-				lineWidth = gridLines.zeroLineWidth;
-				lineColor = gridLines.zeroLineColor;
-				borderDash = gridLines.zeroLineBorderDash || [];
-				borderDashOffset = gridLines.zeroLineBorderDashOffset || 0.0;
-			} else {
-				lineWidth = valueAtIndexOrDefault(gridLines.lineWidth, i, 1);
-				lineColor = valueAtIndexOrDefault(gridLines.color, i, 'rgba(0,0,0,0.1)');
-				borderDash = gridLines.borderDash || [];
-				borderDashOffset = gridLines.borderDashOffset || 0.0;
-			}
+			context = {
+				scale: me,
+				tick,
+			};
+
+			lineWidth = resolve([gridLines.lineWidth, 1], context, i);
```

Comment:
Do we need the second parameter to resolve in each of these cases? They always match the default value. We don't specify the defaults a second time in `_resolveDataElementOptions` and I think it'd be nice to maintain consistency in the way we resolve options

Reply:
I can remove them. Initial goal was minimal changes to see if it worked.